### PR TITLE
Blaze function fix

### PR DESCRIPTION
**Pull request by debora-pe**
This is a small fix to the code that extracts the 1D flat spectrum. Currently, if an archival pixelflat file is used, the `pixelflat_model` image does not exists and the 1D flat spectrum is not extracted. These changes fix this.

### CODE REVIEW

File: pypeit/extraction.py [hunk 1]
```diff
@@ -247,6 +245,18 @@ def __init__(self, sciImg, slits, sobjs_obj, spectrograph, par, objtype, global_
...
             msgs.warn("Spectral FWHM image could not be generated")
 
+        # get flatfield image for blaze function
+        self.flatimg = None
+        if flatimages is not None:
+            # This way of getting the flat image (instead of just reading flatimages.pixelflat_model) ensures that the
+            # flat image is available also when an archival pixel flat is used.
+            flat_raw = flatimages.pixelflat_raw if flatimages.pixelflat_raw is not None else flatimages.illumflat_raw
+            if flat_raw is not None and flatimages.pixelflat_norm is not None:
+                # TODO: Can we just use flat_raw if flatimages.pixelflat_norm is None?
```
Comment (rcooke-ast) [inline pypeit/extraction.py:255]: I think we can, but perhaps add a warning that the raw data will be used?

**Review by rcooke-ast — APPROVED** [review]
All good from me 👍  thanks for the fix!!

One point to note... is there something about archival pixelflats that prevents them from being used (presumably, the archival pixelflats are just the sensitivity deviations about 1)? Do we just need to recreate them with the latest version of the code, or do we need to find an alternative way of storing the archival pixelflats so that they can still be used for extracting the Flatfield spectrum?

**Review by jhennawi — APPROVED** [review]
Looks good.

**Comment by debora-pe** [comment]
> All good from me 👍 thanks for the fix!!
> 
> One point to note... is there something about archival pixelflats that prevents them from being used (presumably, the archival pixelflats are just the sensitivity deviations about 1)? Do we just need to recreate them with the latest version of the code, or do we need to find an alternative way of storing the archival pixelflats so that they can still be used for extracting the Flatfield spectrum?

Thanks @rcooke-ast.
I don't think there is anything in the archival pixelflats that prevent them from being used.  Yes, there are just a pixel to pixel variations about 1 and they are not version controlled, i.e., they do not need to be re-created with the latest version of the code.

File: pypeit/extraction.py (hunk 1 above)
Comment (debora-pe) [reply pypeit/extraction.py:255]: Thanks for your input. I'll leave it as a TODO for now.

**Comment by debora-pe** [comment]
I'll run the tests for completeness.

**Comment by debora-pe** [comment]
Tests pass. See below Summary.
- `UNIT TESTS FAILED  7 failed`  are related to a problem my machine has with setup_gui
-  `PYPEIT UNIT TESTS FAILED  1 failed` was a temporary internet connection problem. It passed today
```
   > pytest ../PypeIt/pypeit/tests/test_install.py::test_install_telluric
../PypeIt/pypeit/tests/test_install.py .                            [100%]
================== 1 passed in 9.19s =============================
```
- `VET TESTS FAILED  1 failed` is unrelated to this PR
   `FAILED ../../../Pypeit/PypeIt-development-suite/vet_tests/test_datacube.py::test_coadd_datacube - pypeit.pypmsgs.PypeItError: No unmasked value in iexp=1/2. Check inputs.`


```
Test Summary
--------------------------------------------------------
--- PYTEST PYPEIT UNIT TESTS FAILED  1 failed, 256 passed, 63 warnings in 439.44s (0:07:19) ---
--- PYTEST UNIT TESTS FAILED  7 failed, 145 passed, 168 warnings in 1742.28s (0:29:02) ---
--- PYTEST VET TESTS FAILED  1 failed, 62 passed, 97 warnings in 2214.09s (0:36:54) ---
--- PYPEIT DEVELOPMENT SUITE PASSED 248/248 TESTS  ---
Testing Started at 2024-11-21T17:49:53.096124
Testing Completed at 2024-11-22T06:58:25.312261
Total Time: 13:08:32.216137
```

**Comment by debora-pe** [comment]
Merging.

**Pull request merged**